exporter/datadogexporter/metrics/utils_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"gopkg.in/zorkian/go-datadog-api.v2"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/config"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/utils/cache"
+)
+
+func TestNewGauge(t *testing.T) {
+	name := "test.metric"
+	ts := uint64(1e9)
+	value := 2.0
+	tags := []string{"tag:value"}
+
+	metric := NewGauge(name, ts, value, tags)
+
+	assert.Equal(t, "test.metric", *metric.Metric)
+	// Assert timestamp conversion from uint64 ns to float64 s
+	assert.Equal(t, 1.0, *metric.Points[0][0])
+	// Assert value
+	assert.Equal(t, 2.0, *metric.Points[0][1])
+	// Assert tags
+	assert.Equal(t, []string{"tag:value"}, metric.Tags)
+}
+
+func TestRunningMetric(t *testing.T) {
+	logger := zap.NewNop()
+	cfg := &config.Config{}
+
+	ms := RunningMetric("metrics", 0, logger, cfg)
+
+	assert.Equal(t, "otel.exporter.metrics.running", *ms[0].Metric)
+	// Assert metrics list length (should be 1)
+	assert.Equal(t, 1, len(ms))
+	// Assert timestamp
+	assert.Equal(t, 0.0, *ms[0].Points[0][0])
+	// Assert value (should always be 1.0)
+	assert.Equal(t, 1.0, *ms[0].Points[0][1])
+}
+
+func TestAddHostname(t *testing.T) {
+	// With hostname in config
+	cache.Cache.Flush()
```

Comment:
Any test that calls `PushMetricsData`, `processMetrics`, `pushTraceData`, `RunningMetric`, `GetHost` or `AddHostname` sets the hostname in the cache. That covers quite a lot of tests (and that number could grow in the future when more tests are added, and it's not always obvious that the cache is used / set), thus I thought it would be simpler to reset the cache when we do need it (as most tests don't care about the state of the cache).

Reply:
Makes sense, let's keep it like it is then. At some point we might want to do the cache per-exporter which would solve this (and allow more flexibility if someone wants to use multiple Datadog exporters for some reason)